shoes-swt/spec/shoes/swt/arc_painter_spec.rb

```diff
@@ -9,72 +9,207 @@
-
-    let(:shape) { Shoes::Swt::Arc.new(dsl, swt_app) }
-    subject { Shoes::Swt::ArcPainter.new(shape) }
-
-    it_behaves_like "stroke painter"
-    it_behaves_like "fill painter"
-
-    context "normal fill style" do
-      before :each do
-        allow(shape).to receive_messages(wedge?: false)
-      end
-
-      specify "fills arc using path" do
-        expect(gc).to receive(:fill_path)
-        subject.paint_control(event)
-      end
-
-      specify "draws arc" do
-        expect(gc).to receive(:draw_arc)
-        subject.paint_control(event)
-      end
-
-      # Swt measures the arc counterclockwise, while Shoes measures it clockwise.
-      specify "translates DSL values for Swt" do
-        path = double('path')
-        allow(::Swt::Path).to receive(:new) { path }
-        args = [100, 200, width, height, 180.0, -90.0]
-        expect(path).to receive(:add_arc).with(*args)
-        sw = 10
-        args = [100 + sw / 2, 200 + sw / 2, width - sw, height - sw, 180, -90.0]
-        expect(gc).to receive(:draw_arc).with(*args)
-        subject.paint_control(gc)
-      end
+      sw = 10
+      args = [100 + sw / 2, 200 + sw / 2, width - sw, height - sw, -180, -270.0]
+      expect(gc).to receive(:fill_path).with(path)
+      expect(gc).to receive(:draw_arc).with(*args)
+      subject.paint_control(gc)
+    end
+  end
+
+  context "wedge fill style" do
+    before :each do
+      allow(shape).to receive_messages(wedge?: true)
     end
 
-    context "wedge fill style" do
-      before :each do
-        allow(shape).to receive_messages(wedge?: true)
-      end
-
-      specify "fills arc" do
-        expect(gc).to receive(:fill_arc)
-        subject.paint_control(event)
-      end
-
-      specify "draws arc" do
-        expect(gc).to receive(:draw_arc)
-        subject.paint_control(event)
-      end
-
-      specify "translates DSL values for Swt" do
-        args = [100, 200, width, height, 180, -90.0]
-        expect(gc).to receive(:fill_arc).with(*args)
-        sw = 10
-        args = [100 + sw / 2, 200 + sw / 2, width - sw, height - sw, 180, -90.0]
-        expect(gc).to receive(:draw_arc).with(*args)
-        subject.paint_control(gc)
-      end
+    specify "fills arc" do
+      expect(gc).to receive(:fill_arc)
+      subject.paint_control(event)
     end
+
+    specify "draws arc" do
+      expect(gc).to receive(:draw_arc)
+      subject.paint_control(event)
+    end
+
+    specify "translates DSL values for Swt" do
+      args = [left, top, width, height, -180, -270.0]
+      expect(gc).to receive(:fill_arc).with(*args)
+
+      sw = 10
+      args = [100 + sw / 2, 200 + sw / 2, width - sw, height - sw, -180, -270.0]
+      expect(gc).to receive(:draw_arc).with(*args)
+      subject.paint_control(gc)
+    end
+  end
+
+  describe "half-way around" do
+    #
+    #   ---------
+    #  /         \
+    # |           |
+    # |...........|
+    # |...........|
+    #  \........./
+    #   ---------
+    #
```

Comment:
I like your little visualizations, also over with text block. I suspected that they do something like you said but was not sure. Maybe a little description about what it's meant to convey above the first one is all we need? :)